src/hotspot/os/linux/os_linux.cpp

```diff
@@ -1786,6 +1786,18 @@ void * os::Linux::dll_load_in_vmthread(const char *filename, char *ebuf,
   return result;
 }
 
+const char* os::Linux::dll_path(void* lib) {
+  struct link_map *lmap;
+  const char* l_path = "<not available>";
```

Comment:
To be honest, I prefer to return NULL if `dlinfo` is failed.
IIUC the caller expects return dll path of `lib` - `<not available>` is not the path.

I think it is better that alternative string would be set by the caller (`os::dll_unload()`) like a change of os_windows. But I do not stick my opinion. I'm ok if other reviewer approve your change.

Reply:
Adjusted dll_path, it returns now NULL in case of failure.